api/src/routes/user.test.ts

```diff
@@ -558,6 +561,105 @@ describe('userRoutes', () => {
         expect(testuser).toStrictEqual(publicUser);
       });
     });
+
+    describe('/user/report-user', () => {
+      let sendEmailSpy: jest.SpyInstance;
+      beforeEach(() => {
+        sendEmailSpy = jest
+          .spyOn(fastifyTestInstance, 'sendEmail')
+          .mockImplementation(jest.fn());
+      });
+
+      afterEach(() => {
+        jest.clearAllMocks();
+      });
+
+      test('POST returns 400 for empty username', async () => {
+        const response = await superRequest('/user/report-user', {
+          method: 'POST',
+          setCookies
+        }).send({
+          username: '',
+          reportDescription: 'Test Report'
+        });
+
+        expect(response.statusCode).toBe(400);
+        expect(response.body).toStrictEqual({
+          type: 'danger',
+          message: 'flash.provide-username'
+        });
+      });
+
+      test('POST returns 400 for empty report', async () => {
+        const response = await superRequest('/user/report-user', {
+          method: 'POST',
+          setCookies
+        }).send({
+          username: 'darth-vader',
+          reportDescription: ''
+        });
+
+        expect(response.statusCode).toBe(400);
+        expect(response.body).toStrictEqual({
+          type: 'danger',
+          message: 'flash.provide-username'
+        });
+      });
+
+      test('POST sanitises report description', async () => {
+        await superRequest('/user/report-user', {
+          method: 'POST',
+          setCookies
+        }).send({
+          username: 'darth-vader',
+          reportDescription:
+            '<script>const breath = "loud"</script>Luke, I am your father'
+        });
+
+        expect(sendEmailSpy).toBeCalledTimes(1);
+        expect(sendEmailSpy).toBeCalledWith(
+          expect.objectContaining({
+            text: expect.stringContaining(
+              'Report Details:\n\nLuke, I am your father'
+            )
+          })
+        );
+      });
+
+      test('POST returns 200 status code with "success" message', async () => {
+        const response = await superRequest('/user/report-user', {
+          method: 'POST',
+          setCookies
+        }).send({
+          username: 'darth-vader',
+          reportDescription: 'Luke, I am your father'
+        });
+
+        const user = await fastifyTestInstance.prisma.user.findFirstOrThrow({
+          where: { email: 'foo@bar.com' }
+        });
+
+        expect(sendEmailSpy).toBeCalledTimes(1);
+        expect(sendEmailSpy).toBeCalledWith({
+          from: 'team@freecodecamp.org',
+          to: 'support@freecodecamp.org',
+          cc: user.email,
+          subject: "Abuse Report: Reporting darth-vader's profile",
+          text: generateReportEmail(
+            user,
+            'darth-vader',
+            'Luke, I am your father'
+          )
```

Comment:
This is slightly circular since we don't test `generateReportEmail`. What I would recommend is to put the text that `generateReportEmail` generates here.